content/spin/sqlite-api-guide.md

```diff
@@ -0,0 +1,184 @@
+* Parameters are JavaScript values (numbers, strings, byte arrays, or nulls). Spin infers the underlying SQL type.
+* The `execute` function returns an object with `rows` and `columns` properties. `columns` is an array of strings representing column names. `rows` is an array of rows, each of which is an array of JavaScript values (as above) in the same order as `columns`.
+* The `Connection` object doesn't surface the `close` function.
+* Errors are surfaced as exceptions.
+
+{{ blockEnd }}
+
+{{ startTab "Python"}}
+
+To use SQLite functions, use the `spin_sqlite` module in the Python SDK. The `sqlite_open` and `sqlite_open_default` functions return a connection object. The connection object provides the `execute` method as described above. For example:
+
+```python
+from spin_http import Response
+from spin_sqlite import sqlite_open_default
+
+def handle_request(request):
+    conn = sqlite_open_default()
+    result = conn.execute("SELECT * FROM todos WHERE id > (?);", [1])
+    rows = result.rows()
+    return Response(200,
+                    {"content-type": "application/json"},
+                    bytes(str(rows), "utf-8"))
+```
+
+**General Notes**
+* Parameters are Python values (numbers, strings, and lists). Spin infers the underlying SQL type.
+* The `execute` method returns an object with `rows` and `columns` methods. `columns` returns a list of strings representing column names. `rows` is an array of rows, each of which is an array of Python values (as above) in the same order as `columns`.
+* The connection object doesn't surface the `close` function.
+* Errors are surfaced as exceptions.
+
+{{ blockEnd }}
+
+{{ startTab "TinyGo"}}
+
+The Go SDK doesn't currently surface the SQLite API.
+
+{{ blockEnd }}
+
+{{ blockEnd }}
+
+## Preparing an SQLite Database
+
+Although Spin provides SQLite as a built-in database, SQLite still needs you to create its tables.  Although you can do this using Spin components and the SQLite API, this usually results in database setup getting uncomfortably mixed in with application logic.
+
+Instead, you can use the `spin up --sqlite` option to run whatever SQL statements you need before your application starts.  This is typically used to create or alter tables, but can be used for whatever other maintenance or troubleshooting tasks you need.
+
+You can run a SQL script from a file using the `@filename` syntax:
+
+<!-- @nocpy -->
+
+```bash
+spin up --sqlite @migration.sql
+```
+
+Or you can pass SQL statements directly on the command line as a (quoted) string:
+
+<!-- @nocpy -->
+
+```bash
+spin up --sqlite "CREATE TABLE IF NOT EXISTS todos (id INTEGER PRIMARY KEY AUTOINCREMENT, description TEXT NOT NULL, due TEXT NOT NULL)"
+```
+
+You can provide the `--sqlite` flag more than once; Spin runs the statements in the order you provide them.
```

Comment:
Does it wait for the first `--sqlite` command to finish before starting the next?